### ISSUE TITLE
Bug: Integration throws error on update when entity is disabled

### ISSUE THREAD

**Issue by monotonus** [JurajNyiri/HomeAssistant-Tapo-Control#32]
**Describe the bug**
I get thousands of these log entries in a few hours.

**To Reproduce**
I just added the camera through integrations page.

**Expected behavior**
No error in log ;)

**Log**
If applicable, add error logs.

```
Logger: custom_components.tapo_control
Source: helpers/entity.py:438
Integration: Tapo: Cameras Control (documentation, issues)
First occurred: 19:26:54 (1094 occurrences)
Last logged: 22:48:07
Unexpected error fetching Tapo resource status data:

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 143, in async_refresh
    self.data = await self._async_update_data()
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 131, in _async_update_data
    return await self.update_method()
  File "/config/custom_components/tapo_control/__init__.py", line 68, in async_update_data
    entity.async_schedule_update_ha_state(True)
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 438, in async_schedule_update_ha_state
    assert self.hass is not None
AssertionError


```

**Camera (please complete the following information):**
 - Device Model: C200
 - FW: 1.0.14 Build 200720 Rel.38552n(4555)

**Using stream component**
yes i think so

**Camera works through onvif component OR different cameras work in general**
yes i think so

**Camera has all attributes filled out properly in developer tools**
yes i think so

**HASS Environment**
Home Assistant Container in Debian VM
Version 0.116.2

**Additional context**
Currently added the camera in a picture-glance card.

**Comment by JurajNyiri** [comment]
I haven't seen this one yet and cannot replicate it.

What the code in tapo is doing is asking Home Assistant to update the entity, it seems to throw an error outside of this integration inside Home Assistant core.

- Does the camera stream work?
- Do services work? For example, changing ptz, turning on led
- Does camera have correct entity attributes?
- If you change some setting via tapo app, for example toggle the led... does it update in attributes for the camera entity in hass? (give it a minute)

**Comment by monotonus** [comment]
*Does the camera stream work?
Yes. With a few seconds delay.

*Do services work? For example, changing ptz, turning on led
Yes. I have automations which are turing the camera to the wall using a prefix, enabling privacy mode and disabling motion detection and vice versa if leaving home.

*Does camera have correct entity attributes?
access_token: ***************************
model_name: C200
brand: TP-Link
motion_detection: off
device_type: SMART.IPCAMERA
device_model: C200
device_name: C200 1.0
device_info: C200 1.0
hw_version: 1.0
sw_version: 1.0.14 Build 200720 Rel.38552n(4555)
device_alias: tapo_camera
features: 3
barcode: 
mac: ***************************
dev_id: ***************************
oem_id: ***************************
hw_desc: ***************************
motion_detection_sensitivity: normal
privacy_mode: off
alarm: off
alarm_mode: sound
led: on
auto_track: off
presets: 
'1': sofa
'2': hinten
'4': esszimmer

*If you change some setting via tapo app, for example toggle the led... does it update in attributes for the camera entity in hass? (give it a minute)
Yes LED state and alarm_mode were changed in the attributes within seconds.

**Comment by monotonus** [comment]
I think I have found the solution.
 
I'm usually disabling entities I don't need and so I disabled the SD stream entity. I deleted the whole integration and added it again and for now don't have disabled the SD stream. No errors so far. I will keep an eye on it and keep you updated.

So maybe it would be a good feature if you have the choice whether you want both streams or just one of them.

**Comment by JurajNyiri** [comment]
I think that delay is just a coincidence, it should not be related at all. Time makes sense, all my cameras have incorrect time because they are disconnected and I haven't yet discovered a way how to update the time via api.

I am going to get a bit technical now...

So the error happens when the integration asks update coordinator (hass helper) to update all entities our integration created. 
Our entity is TapoCamEntity which extends Camera (hass class). Camera extends Entity (hass class). Now this entity should always be aware about hass variable which is always a property of almost every class. 

Your services are working, which means, that our Entity, Camera, TapoCamEntity is aware about hass property and it was passed in correctly by hass when integration was initiated. It is using that property to do some stuff.. if it wasn't passed it wouldn't work. Now for some reason, your Entity does not have a hass property, and it throws that error on assert. Even when that happens, the update which is using the hass property still works, so I am not sure how that is possible.

I will need some help here to figure out what is going on and why it doesn't show up on my 2 different environments (macbook python3 install without virtualenv and HassOS on pi4). Is anyone else experiencing the issue?

**Comment by JurajNyiri** [comment]
I just saw your response as I sent that message. That actually makes sense, I will need to investigate how it all works if entity is disabled. How are you disabling it? I will try to replicate it.

**Comment by monotonus** [comment]
And I also have now seens yours ;)

It's this feature in Home Assistant:

![grafik](https://user-images.githubusercontent.com/51734505/96191059-a1fefc00-0f43-11eb-917d-aedfc2c3dbcd.png)

And of course there is the disclaimer this will not work with every integration :-D

**Comment by JurajNyiri** [comment]
Yeah that makes perfect sense, it explains why Entity wouldn't have hass. Even though we actually try to create it. It would throw that error message on the entity you disabled. I will take a look and try to fix this when I have some time later 👍

**Comment by JurajNyiri** [comment]
Will be fixed in 2.2.
